challenge-2/README.md

```diff
@@ -0,0 +1,52 @@
+# Challenge 2
+Inspired from [Challenge 3 of Advent of Spin 2023](https://github.com/fermyon/advent-of-spin/blob/main/2023/Challenge-3/README.md
+)
+
+
+
+### Spec
+It's summer time! It's all about vacations and enjoying the beautiful weather outdoors. Finally, a group of friends found time to take a trip that they have been putting off for months, Alas! They don't want to deal with planning every part of the trip. Your task is to utilize the LLM powers to help them build a perfect itinerary to enjoy the season. This must be stored in a key-value store for them to refer later as they go on this trip! 
+
+The output generated by the model is not assessed but the http response is, as discussed below.
+
+The users will be able to give some input such as destination, duration, type of trip and number of people. Take the help of LLMs to plan the perfect getaway! The users will also provide a unique `tag` to retrieve the generated result later.
+
+The users will send a POST request to `/plan-my-trip` in JSON format.
+```json
+{
+    "tag": "week-in-banff"
+    "destination": "Banff",
+    "duration": "1 week",
+    "num_people": "four",
+    "activities": ["hiking", "camping", "long drive", "food"]
+    
+}
+```
+
+You will have to create a prompt out the JSON data given along with the request to feed it to the LLM model to generate the itinerary
+Example prompt template might be
+
+```
+PROMPT = " Create a summer vacation detailed itinerary trip to go to <destination> for a <duration>. <num_people> people will be going on this trip planning to do <activities>"
+```
+Do explore different ways to generate the best outcome from the model :D
+
+The expected return type must be a `201` response as follows:
+
+```json
+{
+    "itinerary" : <GENERATED ITINERARY>
+}
```

Comment:
You probably want the backend to generate the tag and not the user providing it to avoid one user overwritting another.